### PR TITLE
Cli infrastucture improvement

### PR DESCRIPTION
**Pull request by vovythevov**
This pull request improves the CLI testing even further by:
 - Simplifying the python CLI tests so they use a CLI that is always available
 - Add a test to look at what signal a CLI might send

### CODE REVIEW

**Comment by vovythevov** [comment]
@jcfr Would you mind taking a quick look ? 
If theere is nothing wrong, I'll merge this tomorrow.

File: Applications/SlicerApp/Testing/Python/CLIEventTest.py [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+    self.ExecutionFinished = False
+    self.StatusEventCallback = None
+
+  def runCLI(self, cliModule, cliNode, parameters, wait_for_completion):
+    self.addObserver(cliNode, self.StatusModifiedEvent, self.onCLIModified)
+    cliNode = slicer.cli.run(cliModule, cliNode, parameters, wait_for_completion)
+
+  def onCLIModified(self, cliNode, event):
+    print "--",cliNode.GetStatusString(),":", cliNode.GetName()
```
Comment (jcfr) [inline Applications/SlicerApp/Testing/Python/CLIEventTest.py]: Nitpicks: use print as a function. `print()`

File: Applications/SlicerApp/Testing/Python/CLIEventTest.py [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+    self.StatusEvents.append(cliNode.GetStatus())
+
+    if not cliNode.IsBusy():
+      self.removeObservers(cliNode, self.StatusModifiedEvent, self.onCLIModified)
+      self.ExecutionFinished = True
+
+    if self.StatusEventCallback:
+      self.StatusEventCallback(cliNode)
+
+  def addObserver(self, object, event, method, group = 'none'):
```
Comment (jcfr) [inline Applications/SlicerApp/Testing/Python/CLIEventTest.py]: Use `VTKObservationMixin`

See http://wiki.slicer.org/slicerWiki/index.php/Documentation/Nightly/Developers/FAQ#How_can_I_access_callData_argument_in_a_VTK_object_observer_callback_function

http://viewvc.slicer.org/viewvc.cgi/Slicer4?view=revision&revision=24569
http://viewvc.slicer.org/viewvc.cgi/Slicer4?view=revision&revision=24570

File: Applications/SlicerApp/Testing/Python/CLIEventTest.py [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+# CLIEventTestLogic
+#
+
+class CLIEventTestTest(ScriptedLoadableModuleTest):
+
+  def setUp(self):
+    """ Reset the state for testing.
+    """
+
+  #def runTest(self):
```
Comment (jcfr) [inline Applications/SlicerApp/Testing/Python/CLIEventTest.py]: Remove comments

File: Applications/SlicerApp/Testing/Python/CLIEventTest.py [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+  def setup(self):
+    ScriptedLoadableModuleWidget.setup(self)
+
+#
+# CLIEventTestLogic
+#
+
+class CLIEventTestLogic(ScriptedLoadableModuleLogic):
+  def __init__(self):
+    self.Observations = []
```
Comment (jcfr) [inline Applications/SlicerApp/Testing/Python/CLIEventTest.py]: In python, private var starts with lower case letter.

**Comment by vovythevov** [comment]
Intergrated in r24857

**Comment by vovythevov** [comment]
Integrated in r24857

**Pull request closed without merging**